File: farm-simulating-game.pl
use strict;
use warnings;

# Define initial values
my $money = 1000;
my $seeds = 10;
my $crops = 0;
my $harvested = 0;

# Loop until user quits
while (1) {
    # Display current status
    print "\nYour farm:\n";
    print "Money: \$$money\n";
    print "Seeds: $seeds\n";
    print "Crops: $crops\n";
    print "Harvested: $harvested\n";

    # Prompt for action
    print "\nWhat would you like to do?\n";
    print "1. Buy seeds (10 seeds for \$10)\n";
    print "2. Plant seeds\n";
    print "3. Harvest crops\n";
    print "4. Sell crops\n";
    print "5. Quit\n";
    print "Enter your choice: ";

    # Get user input
    my $choice = <STDIN>;
    chomp $choice;

    # Handle user input
    if ($choice == 1) {
        if ($money < 10) {
            print "You don't have enough money to buy seeds.\n";
        } else {
            $money -= 10;
            $seeds += 10;
            print "You bought 10 seeds for \$10.\n";
        }
    } elsif ($choice == 2) {
        if ($seeds == 0) {
            print "You don't have any seeds to plant.\n";
        } else {
            $seeds--;
            $crops++;
            print "You planted a seed.\n";
        }
    } elsif ($choice == 3) {
        if ($crops == 0) {
            print "You don't have any crops to harvest.\n";
        } else {
            $crops--;
            $harvested++;
            print "You harvested a crop.\n";
        }
    } elsif ($choice == 4) {
        if ($harvested == 0) {
            print "You don't have any crops to sell.\n";
        } else {
            my $sale_price = $harvested * 2;
            $money += $sale_price;
            $harvested = 0;
            print "You sold your crops for \$$sale_price.\n";
        }
    } elsif ($choice == 5) {
        print "Thanks For Playing!\n";
        last;
    } else {
        print "Invalid choice.\n";
    }
}

File: farm-simulating-game.r
# Text-Based Farm Simulation Game in R

# Define initial variables
crops <- c("corn", "wheat", "potatoes")
crop_count <- c(5, 3, 4)
money <- 1000

# Define function to display current status
display_status <- function() {
  cat(paste0("You currently have ", crop_count, " ", crops, " and $", money, " in the bank.\n"))
}

# Define function to plant crops
plant_crops <- function() {
  # Get user input for crop and quantity
  crop <- readline(prompt = "What crop would you like to plant? (corn/wheat/potatoes) ")
  quantity <- as.integer(readline(prompt = "How many acres would you like to plant? "))
  
  # Check if user has enough money and crops
  cost <- quantity * 100
  crop_index <- match(crop, crops)
  if (cost > money) {
    cat("Sorry, you don't have enough money to plant that many acres.\n")
    return()
  }
  if (quantity > crop_count[crop_index]) {
    cat("Sorry, you don't have that many crops to plant.\n")
    return()
  }
  
  # Update crop count and money
  crop_count[crop_index] <- crop_count[crop_index] - quantity
  money <- money - cost
  
  cat(paste0("You now have ", crop_count[crop_index], " ", crop, "s and $", money, " left in the bank.\n"))
}

# Define function to harvest crops
harvest_crops <- function() {
  # Get user input for crop and quantity
  crop <- readline(prompt = "What crop would you like to harvest? (corn/wheat/potatoes) ")
  quantity <- as.integer(readline(prompt = "How many acres would you like to harvest? "))
  
  # Check if user has enough crops
  crop_index <- match(crop, crops)
  if (quantity > crop_count[crop_index]) {
    cat("Sorry, you don't have that many crops to harvest.\n")
    return()
  }
  
  # Update crop count and money
  yield <- quantity * sample(50:100, 1)
  crop_count[crop_index] <- crop_count[crop_index] - quantity
  money <- money + yield
  
  cat(paste0("You harvested ", yield, " bushels of ", crop, " and now have $", money, " in the bank.\n"))
}

# Define main game loop
while (TRUE) {
  display_status()
  
  # Get user input for action
  action <- readline(prompt = "What would you like to do? (plant/harvest/quit) ")
  
  # Perform action based on user input
  if (action == "plant") {
    plant_crops()
  } else if (action == "harvest") {
    harvest_crops()
  } else if (action == "quit") {
    cat("Thanks for playing!\n")
    break
  } else {
    cat("Invalid action.\n")
  }
}
